File: src/main/java/com/raaghav/clinic/entity/Appointment.java
package com.raaghav.clinic.entity;

import jakarta.persistence.*;

import java.time.LocalDateTime;
import java.time.LocalTime;


@Entity
public class Appointment {
    public enum AppointmentStatus{
        BOOKED,CANCELLED,COMPLETED
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "patient_id")
    private Patient patient;

    @ManyToOne
    @JoinColumn(name ="doctor_id")
    private Doctor doctor;

    @Column(name = "appointment_time")
    private LocalDateTime appointmentTime;

    @Enumerated(EnumType.STRING)
    private AppointmentStatus status;
}

File: frontend/src/types/appointment.ts
export type AppointmentStatus = 'BOOKED' | 'COMPLETED' | 'CANCELLED'

export interface Appointment {
  id: number
  patientId: number
  doctorId: number
  patientName: string
  doctorName: string
  appointmentTime: string
  status: AppointmentStatus
}

export interface AppointmentRequest {
  patientId: number
  doctorId: number
  appointmentTime: string
}

export const APPOINTMENT_STATUSES: AppointmentStatus[] = [
  'BOOKED',
  'COMPLETED',
  'CANCELLED',
]
